spec/services/request_password_reset_spec.rb

```diff
@@ -81,6 +81,52 @@
       end
     end
 
+    context 'when the user is found and confirmed, but is suspended' do
+      subject(:perform) do
+        described_class.new(
+          email: email,
+          irs_attempts_api_tracker: irs_attempts_api_tracker,
+        ).perform
+      end
+
+      before do
+        user.suspend!
+        allow(UserMailer).to receive(:suspended_reset_password).
+          and_wrap_original do |impl, user, email, options|
+            token = options.fetch(:token)
+            expect(token).not_to be_present
+
+            impl.call(user, email, **options)
+          end
+      end
+
+      it 'does not set a password reset token' do
+        expect { subject }.
+          not_to(change { user.reload.reset_password_token })
+      end
+
+      it 'sends the correct email to the suspended user' do
+        allow(UserMailer).to receive(:suspended_reset_password)
+
+        subject
+
+        expect(UserMailer).to have_received(:suspended_reset_password)
```

Comment:
Any ideas on how to expect this properly?  Or any other ideas on how to write this test?

Reply:
Thanks!  It looks like that works!